### ISSUE TITLE
Make static site generation incremental

### ISSUE THREAD

**Issue by apatel762** [apatel762/backlinks-ssg#31]
At the moment, the first thing that happens when you generate the site is to delete all of the HTML files and re-create them from scratch.

This is a _bit_ of a waste when you're only changing a couple of files, or if you have lots of files.

My idea at the moment is to record the runtime of the site generation after you run the `make gen` target in the Makefile. This would be the baseline time, and then the next time you generate the site, the scripts will check when it was that you last ran everything and only re-generate stuff that has been modified since then.

Would be nice to clean up the Makefile as well afterwards so that the `gen` target doesn't depend on the `clean` target.

**Comment by apatel762** [comment]
i created a note about one way of doing this, see `2021-01-29T100946Z.md`

tl;dr:
* store some metadata re: each script run
* look at that metadata upon the next script run
* use SQLite for storage
* treat the db like a cache (generate it if it doesn't exist; script should be runnable without it, albeit slower)

**Comment by apatel762** [comment]
re: the above, part of the metadata that I put into the SQLite db/cache would be a hash of the file. This would give an easy way of checking if the file has changed (just in case we can't rely on the last updated dttm)
